go-controller/pkg/ovn/master.go

```diff
@@ -351,10 +355,57 @@ func (oc *Controller) deleteNode(nodeName string, nodeSubnet *net.IPNet) error {
 	return nil
 }
 
+// As for openshift-sdn, OVN uses an overlay and doesn't need GCE Routes, we need to
+// clear the NetworkUnavailable condition that kubelet adds to initial node
+// status when using GCE (done here: https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/cloud/node_controller.go#L237).
+// See discussion surrounding this here: https://github.com/kubernetes/kubernetes/pull/34398.
+// TODO: make upstream kubelet more flexible with overlays and GCE so this
+// condition doesn't get added for network plugins that don't want it, and then
+// we can remove this function.
+func (oc *Controller) clearInitialNodeNetworkUnavailableCondition(origNode *kapi.Node) {
+	// Informer cache should not be mutated, so get a copy of the object
+	node := origNode.DeepCopy()
+	knode := node
+	cleared := false
+	resultErr := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		var err error
+
+		if knode != node {
+			knode, err = oc.kube.GetNode(node.Name)
+			if err != nil {
+				return err
+			}
+		}
+
+		for i := range knode.Status.Conditions {
+			if knode.Status.Conditions[i].Type == kapi.NodeNetworkUnavailable {
+				condition := &knode.Status.Conditions[i]
+				if condition.Status != kapi.ConditionFalse && condition.Reason == "NoRouteCreated" {
+					condition.Status = kapi.ConditionFalse
+					condition.Reason = "RouteCreated"
+					condition.Message = "ovn-kube cleared kubelet-set NoRouteCreated"
+					condition.LastTransitionTime = metav1.Now()
+					if err = oc.kube.UpdateNodeStatus(knode); err == nil {
+						cleared = true
+					}
+				}
+				break
+			}
+		}
+		return err
+	})
+	if resultErr != nil {
+		logrus.Errorf("status update failed for local node: %v", resultErr)
+	} else if cleared {
+		logrus.Infof("Cleared node NetworkUnavailable/NoRouteCreated condition for %s", node.Name)
+	}
+}
+
 func (oc *Controller) syncNodes(nodes []interface{}) {
 	foundNodes := make(map[string]*kapi.Node)
 	for _, tmp := range nodes {
 		node, ok := tmp.(*kapi.Node)
+		oc.clearInitialNodeNetworkUnavailableCondition(node)
 		if !ok {
```

Comment:
I think we do not need L408. As you can see from the comments at Lines 416-419, for all of the existing nodes we do call onAdd() on it and addNode() gets called.

Reply:
Indeed, I forgot that on L271 we call `onAdd` for all existing objects at startup anyways. I will remove this as you mentioned.